### DISCUSSION TITLE
New3D panel doesn't show grid until data is available (frame error)

### DISCUSSION THREAD

**Discussion by jtbandes** [foxglove/community#527] [Q&A]
Steps to reproduce:
- Load a data source.
- Add a grid.
- Reload the app and do not load a data source.

<img width="459" alt="image" src="https://user-images.githubusercontent.com/14237/177891794-47bd5cd8-cc35-4ed3-8fb5-da95b9a3a3d8.png">

**Comment by snosenzo** [comment]
what's the expected behavior here?
also it looks like it happens with or without a grid whenever a data source isn't specified.

**Comment by jhurliman** [comment]
The current New3D panel is built as a pipeline of datasource -> canvas, so no datasource = empty canvas. It would be nice to rearchitect the panel so rendering is somewhat decoupled from the datasource and we can render grids or other custom (user-added) layers when the app first starts.

I've reclassified this as `ux` following the triaging patterns we've adopted for other quirky behavior such as drag and drop of URDFs. It will be a nice UX improvement, although it's probably a heavy lift to implement.

**Comment by defunctzombie** [comment]
Related: https://github.com/foxglove/community/issues/373

**Comment by defunctzombie** [comment]
One quirk I've noticed when revisiting this is that I cannot select `<Display Frame>` from the Grid settings when I do not have any data loaded. That I believe to be a strange behavior - arguably a bug.
